cpp/src/arrow/compute/kernels/vector_sort_benchmark.cc

```diff
@@ -106,6 +125,19 @@ static void ArraySortFuncBoolBenchmark(benchmark::State& state, const Runner& ru
   ArraySortFuncBenchmark(state, runner, values);
 }
 
+template <typename Runner>
+static void ArraySortFuncStringBenchmark(benchmark::State& state, const Runner& runner,
+                                         int64_t min_length, int64_t max_length) {
+  RegressionArgs args(state);
+
+  const int64_t array_size = args.size / sizeof(int64_t);
```

Comment:
This leads to incorrect bytes/second and items/second reporting. See `RegressionSetArgs`.

Reply:
Hmm OK. IIUC we are parametrizing a benchmark with a bunch of different sized byte streams, the length of which is available through args.size. So is the suggested fix here to divide that by `sizeof(string)`? Or should it be `int32_t` to match the offset of `StringType`?

Reply:
No, if we want a bytes/second metric, we need to give GTest the actual byte size of the string data.

Again, I'll let you look a bit more in detail at what `RegressionSetArgs` does. I'm on vacation currently.

Reply:
Right on - enjoy the time off!

Reply:
Divides `sizeof(int64_t)` is not right. If `args.size` is in bytes, it should divides `(min_length + max_length) / 2`, but must also consider `null_proportions`.

Reply:
So my read on the situation is that RegressionSetArgs is basically parametrizing the test with the different CPU cache sizes plus one that exceeds cache, and we are trying to fit an array into those sizes. That seems easy enough to control with the primitive types, but I may be misunderstanding how the variable length random strings would be guaranteed to fit into that same cache even with `(min_length + max_length) / 2` without querying the size of the array after creating. Also unclear if I need to account for the offsets buffer as part of the cache bounding requirement

Reply:
The cache size-based settings are just an approximation, we don't care if the actual size fits exactly (so @cyb70289's proposed formula is fine).
What's more important is to have accurate figures for bytes/sec and items/sec.

Reply:
As for whether offsets should be taken into account, there's no obviously right answer, but given that we don't count the null bitmap, I'd say we don't _need_ to count the offsets buffer either.

Reply:
Here are the updated values with that change:

```sh
Running ./release/arrow-compute-vector-sort-benchmark
Run on (12 X 4310.45 MHz CPU s)
CPU Caches:
  L1 Data 48 KiB (x6)
  L1 Instruction 32 KiB (x6)
  L2 Unified 1280 KiB (x6)
  L3 Unified 12288 KiB (x1)
Load Average: 7.90, 5.70, 3.53
----------------------------------------------------------------------------------------------------
Benchmark                                          Time             CPU   Iterations UserCounters...
----------------------------------------------------------------------------------------------------
ArraySortIndicesString/49152/10000               249 us          249 us         2985 bytes_per_second=188.346M/s items_per_second=12.3435M/s null_percent=0.01 size=49.152k
ArraySortIndicesString/49152/100                 242 us          242 us         2897 bytes_per_second=194.014M/s items_per_second=12.7149M/s null_percent=1 size=49.152k
ArraySortIndicesString/49152/10                  221 us          221 us         3165 bytes_per_second=212.233M/s items_per_second=13.9089M/s null_percent=10 size=49.152k
ArraySortIndicesString/49152/2                   112 us          112 us         6060 bytes_per_second=418.805M/s items_per_second=27.4468M/s null_percent=50 size=49.152k
ArraySortIndicesString/49152/1                  5.20 us         5.20 us       133594 bytes_per_second=8.79969G/s items_per_second=590.537M/s null_percent=100 size=49.152k
ArraySortIndicesString/49152/0                   240 us          240 us         2826 bytes_per_second=195.617M/s items_per_second=12.82M/s null_percent=0 size=49.152k
ArraySortIndicesString/1048576/100              8447 us         8447 us           83 bytes_per_second=118.391M/s items_per_second=7.7589M/s null_percent=1 size=1048.58k
ArraySortIndicesString/8388608/100             91818 us        91795 us            7 bytes_per_second=87.151M/s items_per_second=5.71153M/s null_percent=1 size=8.38861M
ChunkedArraySortIndicesString/49152/10000        328 us          328 us         2139 bytes_per_second=142.782M/s items_per_second=9.35129M/s null_percent=0.01 size=49.152k
ChunkedArraySortIndicesString/49152/100          319 us          319 us         2179 bytes_per_second=146.763M/s items_per_second=9.61198M/s null_percent=1 size=49.152k
ChunkedArraySortIndicesString/49152/10           284 us          284 us         2433 bytes_per_second=165.067M/s items_per_second=10.8108M/s null_percent=10 size=49.152k
ChunkedArraySortIndicesString/49152/2            152 us          152 us         4687 bytes_per_second=307.783M/s items_per_second=20.1577M/s null_percent=50 size=49.152k
ChunkedArraySortIndicesString/49152/1           6.70 us         6.70 us       104605 bytes_per_second=6.83105G/s items_per_second=458.126M/s null_percent=100 size=49.152k
ChunkedArraySortIndicesString/49152/0            326 us          325 us         1990 bytes_per_second=144.018M/s items_per_second=9.4322M/s null_percent=0 size=49.152k
ChunkedArraySortIndicesString/1048576/100      10470 us        10466 us           69 bytes_per_second=95.5466M/s items_per_second=6.26117M/s null_percent=1 size=1048.58k
ChunkedArraySortIndicesString/8388608/100     110628 us       110623 us            6 bytes_per_second=72.3174M/s items_per_second=4.73932M/s null_percent=1 size=8.38861M
ArrayRankString/49152/10000/tiebreaker:0      267473 ns       267229 ns         2733 bytes_per_second=175.412M/s items_per_second=11.4958M/s null_percent=0.01 size=49.152k
ArrayRankString/49152/10000/tiebreaker:2      245559 ns       245517 ns         2863 bytes_per_second=190.923M/s items_per_second=12.5124M/s null_percent=0.01 size=49.152k
ArrayRankString/49152/10000/tiebreaker:3      266399 ns       266340 ns         2737 bytes_per_second=175.997M/s items_per_second=11.5341M/s null_percent=0.01 size=49.152k
ArrayRankString/49152/100/tiebreaker:0        248463 ns       248425 ns         2845 bytes_per_second=188.689M/s items_per_second=12.3659M/s null_percent=1 size=49.152k
ArrayRankString/49152/100/tiebreaker:2        238895 ns       238844 ns         2936 bytes_per_second=196.257M/s items_per_second=12.8619M/s null_percent=1 size=49.152k
ArrayRankString/49152/100/tiebreaker:3        252202 ns       252153 ns         2851 bytes_per_second=185.899M/s items_per_second=12.1831M/s null_percent=1 size=49.152k
ArrayRankString/49152/10/tiebreaker:0         241138 ns       241029 ns         3056 bytes_per_second=194.479M/s items_per_second=12.7454M/s null_percent=10 size=49.152k
ArrayRankString/49152/10/tiebreaker:2         223276 ns       223249 ns         3172 bytes_per_second=209.968M/s items_per_second=13.7604M/s null_percent=10 size=49.152k
ArrayRankString/49152/10/tiebreaker:3         246466 ns       246129 ns         2786 bytes_per_second=190.449M/s items_per_second=12.4813M/s null_percent=10 size=49.152k
ArrayRankString/49152/2/tiebreaker:0          125514 ns       125467 ns         5796 bytes_per_second=373.605M/s items_per_second=24.4846M/s null_percent=50 size=49.152k
ArrayRankString/49152/2/tiebreaker:2          118292 ns       118211 ns         5230 bytes_per_second=396.538M/s items_per_second=25.9875M/s null_percent=50 size=49.152k
ArrayRankString/49152/2/tiebreaker:3          120854 ns       120825 ns         5912 bytes_per_second=387.958M/s items_per_second=25.4252M/s null_percent=50 size=49.152k
ArrayRankString/49152/1/tiebreaker:0            5764 ns         5753 ns       128137 bytes_per_second=7.95685G/s items_per_second=533.975M/s null_percent=100 size=49.152k
ArrayRankString/49152/1/tiebreaker:2            5499 ns         5498 ns       125044 bytes_per_second=8.32608G/s items_per_second=558.754M/s null_percent=100 size=49.152k
ArrayRankString/49152/1/tiebreaker:3            5357 ns         5355 ns       125042 bytes_per_second=8.54873G/s items_per_second=573.696M/s null_percent=100 size=49.152k
ArrayRankString/49152/0/tiebreaker:0          258138 ns       258109 ns         2766 bytes_per_second=181.609M/s items_per_second=11.9019M/s null_percent=0 size=49.152k
ArrayRankString/49152/0/tiebreaker:2          250541 ns       250474 ns         2792 bytes_per_second=187.145M/s items_per_second=12.2647M/s null_percent=0 size=49.152k
ArrayRankString/49152/0/tiebreaker:3          253012 ns       252992 ns         2776 bytes_per_second=185.283M/s items_per_second=12.1427M/s null_percent=0 size=49.152k
ArrayRankString/1048576/100/tiebreaker:0     8756054 ns      8755400 ns           79 bytes_per_second=114.215M/s items_per_second=7.48521M/s null_percent=1 size=1048.58k
ArrayRankString/1048576/100/tiebreaker:2     8528650 ns      8528126 ns           81 bytes_per_second=117.259M/s items_per_second=7.68469M/s null_percent=1 size=1048.58k
ArrayRankString/1048576/100/tiebreaker:3     8656824 ns      8655985 ns           80 bytes_per_second=115.527M/s items_per_second=7.57118M/s null_percent=1 size=1048.58k
ArrayRankString/8388608/100/tiebreaker:0    96252452 ns     96229618 ns            7 bytes_per_second=83.1345M/s items_per_second=5.4483M/s null_percent=1 size=8.38861M
ArrayRankString/8388608/100/tiebreaker:2    92078251 ns     92057802 ns            7 bytes_per_second=86.9019M/s items_per_second=5.6952M/s null_percent=1 size=8.38861M
ArrayRankString/8388608/100/tiebreaker:3    94816437 ns     94803384 ns            7 bytes_per_second=84.3852M/s items_per_second=5.53027M/s null_percent=1 size=8.38861M
```